File: src/S0801/S0801_v1.cpp
class Solution {
public:
    int minSwap(vector<int>& nums1, vector<int>& nums2) {
        int s1 = 0;
        int s2 = 1;
        for (int i=1;i<nums1.size();i++) {
            if (nums1[i-1]<nums1[i] && nums2[i-1]<nums2[i]) {
                if (nums1[i-1]<nums2[i] && nums2[i-1]<nums1[i]) {
                    s1 = min(s1, s2);
                    s2 = s1 + 1;
                } else { s2 += 1;}
            } else {
                s2 = s2 + s1;
                s1 = s2 - s1;
                s2 = s2 - s1 + 1;
            }
        }
        return min(s1, s2);
    }
};

File: src/S0801/S0801_v1.py
class Solution:
    def minSwap(self, nums1, nums2) -> int:
        s1, s2 = 0, 1
        for i in range(1, len(nums1)):
            if  nums1[i-1]<nums1[i] and nums2[i-1]<nums2[i]:
                if nums1[i-1]<nums2[i] and nums2[i-1]<nums1[i]:
                    s1 = min(s1, s2)
                    s2 = s1 + 1
                else:
                    s2 += 1
            else:
                s1, s2 = s2, s1+1
        return min(s1, s2)
